doc/hotspot-style.md

```diff
@@ -703,6 +701,274 @@ Some relevant sections from cppreference.com:
 Although related, the use of `std::initializer_list` remains forbidden, as
 part of the avoidance of the C++ Standard Library in HotSpot code.
 
+### Local Function Objects
+
+* Local function objects, including lambda expressions, may be used.
+* Lambda expressions must only be used as a downward value.
+* Prefer `[&]` as the capture list of a lambda expression.
```

Comment:
hi, @kimbarrett , 
Why does it sound like a suggestion instead of a mandate? 
You explain why we should use [&] and all your examples are implicit capture reference.

Reply:
That's intentional. Capture lists were by far the single most contentious part
of the Oracle-internal discussions about lambda usage. Making this a
suggestion rather than a mandate, along with the rationale, leaves open the
possibility of using other forms *with good reason*, and preferably after a
period of folks getting used to the "usual" form. There are HotSpot developers
who aren't familiar with lambda-like constructs, or are not comfortable with
the introduction of yet more non-alphanumeric syntax that resembles line
noise; limiting the capture list features being used helps them. On the other
hand, there have been proponents of nearly every capture list feature for some
use case or another.

Reply:
Got it. Thanks for the explanation.